velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -476,19 +482,30 @@ void ReaderBase::scheduleRowGroups(
       currentGroup + 1 < rowGroupIds.size() ? rowGroupIds[currentGroup + 1] : 0;
   auto input = inputs_[thisGroup].get();
   if (!input) {
-    auto newInput = input_->clone();
-    reader.enqueueRowGroup(thisGroup, *newInput);
-    newInput->load(dwio::common::LogType::STRIPE);
-    inputs_[thisGroup] = std::move(newInput);
+    if (preloadFile_) {
+      // Read data from buffer directly.
+      reader.enqueueRowGroup(thisGroup, *input_);
+      inputs_[thisGroup] = input_;
+    } else {
+      auto newInput = input_->clone();
+      reader.enqueueRowGroup(thisGroup, *newInput);
+      newInput->load(dwio::common::LogType::STRIPE);
+      inputs_[thisGroup] = std::move(newInput);
+    }
   }
   for (auto counter = 0; counter < FLAGS_parquet_prefetch_rowgroups;
        ++counter) {
     if (nextGroup) {
-      if (inputs_.count(nextGroup) != 0) {
-        auto newInput = input_->clone();
-        reader.enqueueRowGroup(nextGroup, *newInput);
-        newInput->load(dwio::common::LogType::STRIPE);
-        inputs_[nextGroup] = std::move(newInput);
+      if (inputs_.count(nextGroup) == 0) {
```

Comment:
Looks like you fixed a bug in the row group prefetching. We should fix this independently along with a test. Can you open an issue and undo this fix for this PR?

Reply:
Separated: https://github.com/facebookincubator/velox/pull/6121